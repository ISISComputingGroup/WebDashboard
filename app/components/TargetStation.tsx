import InstrumentWallCard from "./InstrumentWallCard";

import { instListEntryWithRunstatePVandValue } from "@/app/types";

export default function TargetStation({
  name,
  instruments,
  beamCurrent,
}: {
  name: string;
  instruments: Array<instListEntryWithRunstatePVandValue>;
  beamCurrent: number | null | undefined;
}) {
  instruments = instruments.sort((instrumenta, instrumentb) =>
    instrumenta.name.localeCompare(instrumentb.name),
  );
  return (
    <div className="flex flex-col justify-center items-start w-full">
      <h1 className="w-full text-left text-gray-600 dark:text-gray-200 font-semibold text-md mt-2 py-2 ">
        {name}{" "}
        {beamCurrent !== undefined && beamCurrent !== null
          ? "- " + beamCurrent.toFixed(2) + " Î¼A"
          : ""}
      </h1>
      <div className="flex flex-wrap gap-1">
        {instruments.map((instrument) => (
          <InstrumentWallCard key={instrument.name} instrument={instrument} />
        ))}
      </div>
    </div>
  );
}
