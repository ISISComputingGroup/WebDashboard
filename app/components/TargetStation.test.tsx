import { render } from "@testing-library/react";
import TargetStation from "@/app/components/TargetStation";

it("renders targetstation unchanged", () => {
  const { container } = render(
    <TargetStation
      name={"TS5"}
      beamCurrent={1234.5678}
      instruments={[
        { name: "Instrument", runStateValue: "RESUMING", pvPrefix:"",runStatePV:"",groups:["EXCITATIONS"], isScheduled:true,seci:false,hostName:"",targetStation:"TS5" },
        {
          name: "Instrument2",
          runStateValue: "VETOING", pvPrefix:"",runStatePV:"",groups:["EXCITATIONS"], isScheduled:true,seci:false,hostName:"",targetStation:"TS5"
        },
      ]}
    />,
  );
  expect(container).toMatchSnapshot();
});
