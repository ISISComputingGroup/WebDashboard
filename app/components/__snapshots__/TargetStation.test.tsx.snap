// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders targetstation unchanged 1`] = `
<div>
  <div
    class="flex flex-col justify-center items-start w-full"
  >
    <h1
      class="w-full text-left text-gray-600 dark:text-gray-200 font-semibold text-md mt-2 py-2 "
    >
      TS5
    </h1>
    <div
      class="flex flex-wrap gap-1"
    >
      <div
        class="flex"
      >
        <a
          class="flex items-center justify-center text-center p-3 w-36 max-h-12 rounded-lg shadow-sm border-2 border-gray-700 dark:border-gray-200 hover:shadow-lg hover:border-black dark:hover:border-gray-700 transition-all duration-200
      bg-green-900 text-white"
          href="/instrument?name=Instrument"
          target="_blank"
        >
          <div
            class="flex flex-col"
          >
            <span
              class="text-sm font-bold line-clamp-1 w-full"
            >
              Instrument
            </span>
            <span
              class="text-xs "
            >
              RESUMING
            </span>
          </div>
        </a>
      </div>
      <div
        class="flex"
      >
        <a
          class="flex items-center justify-center text-center p-3 w-36 max-h-12 rounded-lg shadow-sm border-2 border-gray-700 dark:border-gray-200 hover:shadow-lg hover:border-black dark:hover:border-gray-700 transition-all duration-200
      bg-[#FFFF00] text-black"
          href="/instrument?name=Instrument2"
          target="_blank"
        >
          <div
            class="flex flex-col"
          >
            <span
              class="text-sm font-bold line-clamp-1 w-full"
            >
              Instrument2
            </span>
            <span
              class="text-xs "
            >
              VETOING
            </span>
          </div>
        </a>
      </div>
    </div>
  </div>
</div>
`;
